exporter/prometheusexporter/prometheus.go

```diff
@@ -63,11 +63,17 @@ func (pe *prometheusExporter) Start(ctx context.Context, host component.Host) er
 		return err
 	}
 
-	pe.shutdownFunc = ln.Close
-
 	mux := http.NewServeMux()
 	mux.Handle("/metrics", pe.handler)
 	srv, err := pe.config.ToServer(ctx, host, pe.settings, mux)
+	pe.shutdownFunc = func(ctx context.Context) error {
+		err := ln.Close()
```

Comment:
looks like this is failing the linter check since err shadows the err variable outside of the function.  consider renaming this err

Reply:
Sorry about that, renamed both errors, as err2 also wasn't the prettiest of names